codec/aminojson/json_marshal.go

```diff
@@ -0,0 +1,247 @@
+package aminojson
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	cosmos_proto "github.com/cosmos/cosmos-proto"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+
+	"cosmossdk.io/api/amino"
+)
+
+type JSONMarshaller interface {
+	MarshalAmino(proto.Message) ([]byte, error)
+}
+
+type AminoJSON struct {
+	// maps cosmos_proto.scalar -> zero value factory
+	zeroValues      map[string]func() protoreflect.Value
```

Comment:
Why a zero value factory? How about when a scalar field is set to something non-zero?

Reply:
When a `cosmos_proto.scalar` annotated field is set to something non-zero I'm just marshaling according to the proto/json primitive.  For example:
https://github.com/cosmos/cosmos-sdk/blob/ca28ffcc52f0972ef1d348c2b39edde2331775be/codec/aminojson_test.go#L36-L38

Stake is typed as a string in pulsar but a `types.Dec` in gogo.  Since this encoder only supports pulsar types I'm just using the default behavior.   Wherever Stake was set would need to ensure the equivalence expressed in this test case.

Does this reasoning make sense? I guess the alternative would be for this encoder to go from `proto primitive (e.g. string) -> custom sdk type -> bytes`, but is that possible in all cases?